activerecord/test/cases/enum_test.rb

```diff
@@ -508,4 +508,24 @@ def self.name; "Book"; end
   test "data type of Enum type" do
     assert_equal :integer, Book.type_for_attribute("status").type
   end
+
+  test "enum with string attribute defined as hash" do
+    klass = Class.new(ActiveRecord::Base) do
+      self.table_name = "books"
+      enum cover: { hardcover: :hardcover, paperback: :paperback }
```

Comment:
We could use `enum cover: %i(hardcover paperback).index_by(&:itself)` for the case.

Reply:
Yes. You're right. I decided to use this form for better clarity.